### DISCUSSION TITLE
目前 cache 和 lazy_cache 是否真正起作用？

### DISCUSSION THREAD

**Discussion by Henry-ZHR** [IrineSistiana/mosdns#841] [Q&A]
https://github.com/IrineSistiana/mosdns/blob/025823c1a5c9753c0e35f8b444b9e3c4472d37d1/plugin/executable/cache/cache.go#L209

我没理解错的话，目前是无论是否命中缓存都会执行后续序列，cache 所提供的 response 会被覆盖

**Comment by Henry-ZHR** [comment]
有 `has_resp` 可以直接 accept

但是 wiki 的配置示例里没有提及，cache 里也没有说会执行后续操作，只能说有误导性

**Reply by Sereinfy** [reply]
`"elapsed": "0s"`  应该是找到缓存了，然后直接accept，但还是打印了下一句query_summary
我是写的问题吗
```
      - exec: $cache
      - exec: jump has_resp_sequence
      - exec: query_summary
```

**Comment by tlze** [comment]
exec: jump has_resp_sequence

jump就是会返回才正常，不要返回应该用goto，但无论用那个，感觉你这写法都好奇怪。

**Reply by Sereinfy** [reply]
https://github.com/sbwml/luci-app-mosdns/tree/v5/luci-app-mosdns，去看下luci 作者的写法。你觉得奇怪，和写法本身奇怪，是两回事。jump 过去是加了 accept 。不会回来了。（可能）

**Reply by Sereinfy** [reply]
你在这个讨论区搜一下“配置” 有 6 个，一半是这么写的。再加上luci 作者这么写，应该没啥问题。。（我也不确定）
